transform/convert_quadkey_and_Vertical_id.go

```diff
@@ -576,6 +664,97 @@ func convertVerticallIDToBit(vZoom int64, vIndex int64, outputZoom int64, maxHei
 
 }
 
+// transforms the vertical index with a given index-ZoomLevel pair and returns the vertical index(es) that occupy the same vertical space with a new zoomLevel, zoomScalar, and vertical offset.
+//
+// input:
+//
+//	inputIndex: the index to transform
+//
+//	inputZoom: the zoomLevel of the inputIndex
+//
+//	outputZoom: the zoomlevel of the outputIndex. Determines the number of indicies between min and max altitudes in the output.
+//
+//	altitudeRangeScalar: reduces the difference between min and max altitudes
+//
+//	indexOffset: an integer to shift the height of the inputIndex up or down. A positive offset means the transformed index is higher in altitude than that of the input.
+//
+// output:
+//
+//	[]int64 list of vertical index(es), error
+func convertVerticalIndex(inputIndex int64, inputZoom int64, outputZoom int64, altitudeRangeScalar int64, indexOffset int64) ([]int64, error) {
+
+	var (
+		outputIndexes []int64
+		error         error
+	)
+
+	// determine the upper and lower index bounds to search for matches in height solution space
+	lowerBound, error := calculateMinVerticalIndex(inputIndex, inputZoom, outputZoom, altitudeRangeScalar, indexOffset)
+	if error != nil {
+		return nil, error
+	}
+	upperBound, error := calculateMinVerticalIndex(inputIndex+1, inputZoom, outputZoom, altitudeRangeScalar, indexOffset)
+	if error != nil {
+		return nil, error
+	}
+
+	// Determine the vertical index/indices to return.
+	// a) always return the lowerBound index. Regardless of the difference between the inputZoom and outputZoom,
+	// mathematically the altitude associated with the lower bounds will always satisfy the solution set.
+	// b) cycle through indices from lowerBounds+1 to upperBounds with i to find any possible additional indexes
+	// that satisfy the solution set.
+	outputIndexes = append(outputIndexes, lowerBound)
+
+	for i := lowerBound + 1; i < upperBound; i++ {
+
+		outputIndexes = append(outputIndexes, int64(i))
+	}
+
+	return outputIndexes, nil
+
+}
+
+// converts a vertical index from one set of zoom parameters to another disregarding the floor() cacluation. This creates a simplier system of equations where the solu	tion set for height is a single variable. However, this does not describe the full solution set of height since we have excluded the floor calculation; it describes the condition where m = x, given m = floor(x) if and only if m <= x < m +1;
+func calculateMinVerticalIndex(inputIndex int64, inputZoom int64, outputZoom int64, altitudeRangeScalar int64, indexOffset int64) (int64, error) {
+
+	// check to make sure the input index exists in the input world
+	inputResolution := CalculateArithmeticShift(1, inputZoom)
+
+	maxInputIndex := int64(inputResolution - 1)
+	minInputIndex := -int64(inputResolution + 1)
```

Comment:
If you set inputZoom as 0, this minInputIndex is -2.
But the correct minInputIndex is -1 as 18 page of [the specification]( https://www.ipa.go.jp/digital/architecture/Individual-link/nq6ept000000g0fh-att/4dspatio-temporal-guideline-gamma.pdf).
And inputResolution is int64, so the type conversion is  wasteful.
Modify this to `	minInputIndex := -inputResolution`.
The same applies the others.